### PR TITLE
Ctw 328/add practitioner fetch

### PR DESCRIPTION
**Pull request by luthfib**
Add display when saving the recorder field for condition.

### CODE REVIEW

File: src/fhir/practitioner.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+      "Practitioner",
+      fhirClient,
+      {
+        _id: practitionerId,
+      }
+    );
+
+    return practitioners;
+  } catch (e) {
+    throw new Error(`Failed fetching practitioner: ${e}`);
```
Comment (pmadria3) [inline src/fhir/practitioner.ts:19]: For this error maybe we can also include the practitionerID as well

File: src/fhir/practitioner.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
+import Client from "fhir-kit-client";
+import { searchBuilderRecords } from "./search-helpers";
+
+export const getPractitioner = async (
```
Comment (WesleyKapow) [inline src/fhir/practitioner.ts:4]: Change to return only one and error if 0 found.

**Review by WesleyKapow — CHANGES_REQUESTED** [review]
Minor comments, LGTM

**Review by pmadria3 — APPROVED** [review]
LGTM

File: src/fhir/practitioner.ts [hunk 3]
```diff
@@ -14,7 +14,11 @@ export const getPractitioner = async (
       }
     );
 
-    return practitioners;
+    if (practitioners.length === 0) {
+      throw new Error(`No practitioner found with an id of: ${practitionerId}`);
+    }
```
Comment (WesleyKapow) [inline src/fhir/practitioner.ts:19]: I've hit this tricky problem before. This `throw` will get caught in the below `catch`.

**Review by WesleyKapow — CHANGES_REQUESTED** [review]

File: src/fhir/practitioner.ts (hunk 3 above)
Comment (luthfib) [reply src/fhir/practitioner.ts]: You are right, good catch.

**Review by WesleyKapow — DISMISSED** [review]
LGTM, may want to push the use of `new PractitionerModel` into the helper as well. I think it makes sense for our fhir helpers to just return the models since that's mostly what we want to be working with.

File: src/components/content/forms/conditions.ts [hunk 4]
```diff
@@ -48,8 +47,7 @@ export const conditionSchema = z.object({
 });
 
 const setRecorderField = async (practitionerId: string, fhirClient: Client) => {
-  const practitioner = await getPractitioner(practitionerId, fhirClient);
-  const practitionerModel = new PractitionerModel(practitioner);
+  const practitionerModel = await getPractitioner(practitionerId, fhirClient);
   const display = practitionerModel.fullName;
```
Comment (WesleyKapow) [inline src/components/content/forms/conditions.ts:51]: can simplify to just calling it a `practitioner` and not model.

**Review by WesleyKapow — APPROVED** [review]

**Pull request merged**